### PR TITLE
Add `openTokenBatch` collection to optimize database queries.

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: lib/tokens.js [hunk 1]
```diff
@@ -40,25 +40,31 @@ const idEncoder = new IdEncoder({
...
   ]);
 
   await promisify(database.createIndexes)([{
-    // token batches are sharded by `tokenBatch.id` -- can't shard by
-    // `internalId` because it will not be known at query time because it is
-    // not shared externally and can't make this index unique because the
-    // collection is sharded on a field that isn't part of it; however,
-    // uniqueness is handled by the index below and by the randomness property
-    // of batch IDs
+    /* Note: The `tokenBatch` collection is sharded by `tokenBatch.id`. This
```
Comment (mattcollier) [inline lib/tokens.js:48]: this might read 'should be sharded' since the shards are setup outside of code here.

File: lib/tokens.js [hunk 2]
```diff
@@ -80,7 +86,8 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
     options: {unique: true, background: false}
   }]);
 
-  // FIXME: create `tokenBBatch.id` shard key
+  // FIXME: create `tokenBatch.id` shard key on `tokenBatch` and
```
Comment (mattcollier) [inline lib/tokens.js:89]: does this imply that we hope to implement sharding in code here?  I don't know if that's a thing and if it is if it's something we would want to do at this level.

Comment (dlongley) [reply lib/tokens.js:89]: @mattcollier yes, it implies that hope -- I also don't know if it's a thing.

Comment (mattcollier) [reply lib/tokens.js:89]: k, I'll create an issue to investigate the idea of sharding in code, if we can remove the fixme that would be great

**Pull request merged**